src/main/java/com/avioconsulting/mule/connector/vault/provider/internal/vault/client/auth/algorithm/AWSV4SignProperties.java

```diff
@@ -0,0 +1,96 @@
+package com.avioconsulting.mule.connector.vault.provider.internal.vault.client.auth.algorithm;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AWSV4SignProperties {
+
+    private static final String DEFAULT_HOST = "sts.amazonaws.com";
+    private static final String DEFAULT_SERVICE_NAME = "sts";
+    private static final String DEFAULT_REGION = "us-east-1";
+    private static final String DEFAULT_METHOD = "POST";
+    private static final String DEFAULT_PAYLOAD = "Action=GetCallerIdentity&Version=2011-06-15";
+    private static final String DEFAULT_CANONICAL_URI = "/";
+
+    private String region = DEFAULT_REGION;
+    private String serviceName = DEFAULT_SERVICE_NAME;
+    private String method = DEFAULT_METHOD;
+    private String host;
+    private String canonicalUri;
+    private Map<String, String> queryParameters;
+
+    public AWSV4SignProperties(String url) {
+        process(url);
+    }
+
+    public String getRegion() {
+        return region;
+    }
+
+    public String getServiceName() {
+        return serviceName;
+    }
+
+    public String getMethod() {
+        return method;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public String getCanonicalUri() {
+        return canonicalUri;
+    }
+
+    public Map<String, String> getQueryParameters() {
+        return queryParameters;
+    }
+
+    private void process(String url) {
+        URI uri;
+        try {
+            uri = new URI(url);
+            host = uri.getHost();
+            canonicalUri = uri.getPath();
+            queryParameters = splitQuery(uri.getQuery());
+        } catch (URISyntaxException e) {
```

Comment:
I think we want to at least log that this is happening so we can understand why the defaults are being used when the user is entering an URL. Otherwise we should throw the exception so the URL can be fixed.